File: src/2441.largest-positive-integer-that-exists-with-its-negative.2524/solution.ts
function findMaxK (nums: number[]): number {
    let result = -1;
    const set = new Set(nums);
    for (const num of set) {
        if (num < 0) {
            continue;
        }
        if (set.has(-num)) {
            result = Math.max(result, num);
        }
    }
    return result;
}

File: src/2441.largest-positive-integer-that-exists-with-its-negative.2524/solution.java
class Solution {
    public int findMaxK(int[] nums) {
        int result = -1;
        Set<Integer> set = new HashSet<>();
        for (int n : nums) {
            if (set.contains(-n)) {
                result = Math.max(result, Math.abs(n));
            }
            set.add(n);
        }
        return result;
    }
}